js/src/components/AdvancedSettings.js

```diff
@@ -0,0 +1,194 @@
+import Collapsible from "./SidebarCollapsible";
+import { __, sprintf } from "@wordpress/i18n";
+import { MultiSelect, Select } from "@yoast/components/src/select/Select";
+import RadioButtonGroup from "@yoast/components/src/radiobutton/RadioButtonGroup";
+import TextInput from "@yoast/components/src/inputs/TextInput";
+import { curryUpdateToHiddenInput, getValueFromHiddenInput } from "@yoast/helpers";
+import { Component } from "@wordpress/element";
+import { Fragment } from "react";
+import { Alert } from "@yoast/components";
+
+/**
+ * Function to check whether the current object refers to a post or a taxonomy.
+ *
+ * @returns {boolean} true if it is a post, false otherwise.
+ */
+const isPost = () => window.wpseoAdminL10n.postType === "post";
+
+/**
+ * The values that are used for the noIndex field differ for posts and taxonomies. This function returns an array of
+ * options that can be used to populate a select field.
+ *
+ * @returns {void} Array Returns an array of options for the noIndex setting.
+ */
+const getNoIndexOptions = () => {
+	const noIndex = window.wpseoAdminL10n.noIndex ? "No" : "Yes";
```

Comment:
"No" and "Yes" need to be translated, maybe reuse the translations in the options as well.